src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +280,839 @@ static DamageTable D_808792D4 = {
+    } else {
+        sp80 = &D_80879334;
+        sp7C = &D_80879338;
+    }
+
+    temp1 = this->unk_334 * 300.0f;
+    temp2 = this->unk_334 * 15.0f;
+    temp3 = this->unk_334 * 30.0f;
+
+    for (i = 0; i < 3; i++) {
+        sp88.x = randPlusMinusPoint5Scaled(temp3) + arg2->x;
+        sp88.y = Rand_ZeroFloat(5.0f) + this->actor.floorHeight;
+        sp88.z = randPlusMinusPoint5Scaled(temp3) + arg2->z;
+        D_8087933C.x = randPlusMinusPoint5Scaled(2.0f);
+        D_8087933C.z = randPlusMinusPoint5Scaled(2.0f);
+        func_800B0DE0(globalCtx, &sp88, &D_801D15B0, &D_8087933C, sp80, sp7C, temp1, temp2);
+    }
+}
+
+void func_80876B08(EnDodongo* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 yDiff = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+    if (ABS_ALT(yDiff) < 0x4000) {
+        if (!(player->stateFlags1 & 0x00800000) && (Player_GetMask(globalCtx) != PLAYER_MASK_STONE_MASK)) {
+            func_808777A8(this);
+        } else {
+            func_80877494(this);
+        }
+    } else if (Player_GetMask(globalCtx) != PLAYER_MASK_STONE_MASK) {
+        func_80878354(this);
+    } else {
+        func_80877494(this);
+    }
+}
+
+void func_80876BD0(EnDodongo* this, GlobalContext* globalCtx, s32 arg2) {
+    if (this->actor.colChkInfo.damageEffect == 2) {
+        this->unk_300 = 0;
+        this->unk_340 = 0.75f;
+        this->unk_33C = 4.0f;
+    } else if (this->actor.colChkInfo.damageEffect == 4) {
+        this->unk_300 = 20;
+        this->unk_340 = 0.75f;
+        this->unk_33C = 4.0f;
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG,
+                    this->collider1.elements[arg2].info.bumper.hitPos.x,
+                    this->collider1.elements[arg2].info.bumper.hitPos.y,
+                    this->collider1.elements[arg2].info.bumper.hitPos.z, 0, 0, 0, CLEAR_TAG_LARGE_LIGHT_RAYS);
+    }
+}
+
+void func_80876CAC(EnDodongo* this) {
+    this->unk_300 = 10;
+    this->collider1.base.colType = 3;
+    this->unk_340 = 0.75f;
+    this->unk_344 = 1.125f;
+    this->unk_33C = 1.0f;
+    this->unk_302 = 80;
+    this->actor.flags &= ~0x400;
+    func_800BCB70(&this->actor, 0x4000, 0xFF, 0, 80);
+}
+
+void func_80876D28(EnDodongo* this, GlobalContext* globalCtx) {
+    if (this->unk_300 == 10) {
+        this->unk_302 = 0;
+        this->actor.colorFilterTimer = 0;
+        this->unk_300 = 0;
+        this->collider1.base.colType = 0;
+        this->unk_33C = 0.0f;
+        func_800BF7CC(globalCtx, &this->actor, &this->unk_348[0], 9, 2, this->unk_334 * 0.3f, this->unk_334 * 0.2f);
+        this->actor.flags |= 0x400;
+    }
+}
+
+void func_80876DC4(EnDodongo* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f sp80;
+    Vec3f sp74;
+    Vec3f sp68;
+    s16 sp66;
+    s16 sp64;
+    s16 sp62;
+    f32 temp_f22;
+    f32 temp_f20;
+
+    Math_Vec3f_Copy(&sp68, &this->unk_348[0]);
+    sp66 = (Rand_Next() >> 0x12) + this->actor.shape.rot.y;
```

Comment:
```suggestion
    sp66 = (Rand_Next() / 0x40000) + this->actor.shape.rot.y;
```
Does this match?

Reply:
Unfortunately not, I tried this originally, but it's different code gen.